docs/usage/cloud.rst

```diff
@@ -0,0 +1,139 @@
+.. usage/cloud:
+
+Cloud management
+================
+
+We support training in `Google Cloud ML engine <https://cloud.google.com/ml-engine/>`_,
+which has native Tensorflow support.
+
+Instead of building Python packages yourself and using Google Cloud SDK, we baked
+the process inside Luminoth itself, so you can pull it of with a few simple commands.
+
+You can choose how many workers you want, which `scale tiers
+<https://cloud.google.com/ml-engine/docs/concepts/training-overview#scale_tier>`_
+to use, and where to store the results. We also provide some utilities to
+monitor and manage your job right from your command line.
+
+For all the cloud functionalities, the files read (such as the datasets in
+TFRecord format) and written (such as logs and checkpoints) by Luminoth will reside
+in `Buckets <https://cloud.google.com/storage/docs/creating-buckets>`_ instead of your
+local disk.
+
+Pre-requisites
+``````````````
+
+#. Create a `Google Cloud project <https://console.cloud.google.com/projectcreate>`_.
+#. Install `Google Cloud SDK <https://cloud.google.com/sdk/>`_ on your machine.
+   Although it is not strictly necessary, it will be useful to enable the required
+   APIs and upload your dataset.
+#. Login with Google Cloud SDK::
+
+    gcloud auth login
+
+#. Enable the following APIs:
+     * Compute Engine
+     * Cloud Machine Learning Engine
+     * Google Cloud Storage
+   You can do it through the `web console <https://support.google.com/cloud/answer/6158841>`_
+   or with the following command::
+
+     gcloud services enable compute.googleapis.com ml.googleapis.com storage-component.googleapis.com
+
+   Don't worry, it can take a few minutes!
+
+#. Upload your dataset's TFRecord files to a Cloud Storage bucket::
+
+    gsutil -o GSUtil:parallel_composite_upload_threshold=150M cp -r /path/to/dataset/tfrecords gs://your_bucket/path
+
+#. Create a `Service Account Key <https://console.cloud.google.com/iam-admin/serviceaccounts/project>`_
+   (JSON format) and download it to your directory of choice. You may add it as an Editor of your project.
+   If necessary, add required roles (permissions) to your service account.
+
+#. Point the ``GOOGLE_APPLICATION_CREDENTIALS`` environment variable to the JSON file of the service account.
+
+Running a training job
+``````````````````````
+
+Luminoth command line tool provides commands to submit training jobs, list them
+and fetch their logs.
+
+``lumi cloud gc train`` - Submit a training job.
+
+Required arguments:
+  - ``--config``: YAML configuration file for use in training.
+
+Optional arguments:
+  - ``--dataset``: full path to bucket with the dataset's TFRecord files, ie.
```

Comment:
Do we have a reason to keep this config option instead of having e.g. `-o dataset.dir=gs://<bucket_name>/<path>`?

Reply:
It can be done like that, yes.

Overriding that is what it will end up doing. But maybe it's more clear to be using `--dataset` because here it always points to a path in a bucket, while `dataset.dir` might also be local files.